modular_skyrat/modules/icspawning/code/observer.dm

```diff
@@ -1,105 +1,115 @@
-		var/list/outfits = list()
-		outfits["Bluespace Tech"] = /datum/outfit/debug/bst
-		outfits["Bluespace Tech (MODsuit)"] = /datum/outfit/admin/bst
-		outfits["Show All"] = "Show All"
-
-		var/dresscode
-		var/teleport_option = tgui_alert(usr, "How would you like to be spawned in?", "IC Quick Spawn", list("Bluespace", "Pod", "Cancel"))
-		if (teleport_option == "Cancel")
+/mob/dead/observer/CtrlClickOn(mob/target)
+	quickicspawn(target, src)
+
+/mob/dead/observer/proc/quickicspawn(mob/target, mob/user)
+	if(!isobserver(target) || !check_rights(R_SPAWN))
+		return
+
+	/// Whether to spawn in with sparks or in a pod
+	var/teleport_option
+	/// What style of pod to use, if the option was chosen
+	var/pod_style
+	/// Assoc list of pod UI names to style datums, static so it's only built once
+	var/static/list/pod_styles
+	/// Whether to spawn in as your current character or a random one
+	var/character_option
+	/// Which outfit to use
+	var/outfit_option
+	/// Initial list of outfits
+	var/list/outfits = list(
+		"Bluespace Tech" = /datum/outfit/admin/bst,
+		"Naked" = /datum/outfit,
+		"Show All" = "Show All",
+	)
+	/// Whether to grant the return spell
+	var/give_return
+	/// Whether to spawn with quirks and/or loadout
+	var/give_quirks_loadout
+
+	teleport_option = tgui_alert(user, "How would you like to be spawned in?", "IC Quick Spawn", list("Bluespace", "Pod", "Cancel"))
+	if(!teleport_option || teleport_option == "Cancel")
+		return
+
+	if(teleport_option == "Pod")
+		if(!pod_styles)
+			pod_styles = list()
+			for(var/datum/pod_style/style as anything in typesof(/datum/pod_style))
+				pod_styles[style::ui_name] = style
+
+		pod_style = tgui_input_list(user, "Which style of pod?", "IC Quick Spawn", pod_styles)
+		pod_style = pod_styles[pod_style]
+		if(!pod_style)
 			return
-		var/character_option = tgui_alert(usr, "Which character?", "IC Quick Spawn", list("Selected Character", "Randomly Created", "Cancel"))
-		if (character_option == "Cancel")
+
+	character_option = tgui_alert(user, "Which character to spawn as?", "IC Quick Spawn", list("Selected Character", "Random Character", "Cancel"))
+	if(!character_option || character_option == "Cancel")
+		return
+
+	outfit_option = tgui_input_list(user, "Which outfit to use?", "IC Quick Spawn", outfits)
+	if(outfit_option == "Show All")
+		outfit_option = user.client.robust_dress_shop_skyrat()
+	else
+		outfit_option = outfits[outfit_option]
+	if(!outfit_option)
+		return
+
+	// The give return option is only relevant if the user is spawning in someone else
+	if(target != user)
+		give_return = tgui_alert(user, "Do you want to give them the power to return? Not recommended for non-admins.", "IC Quick Spawn", list("Yes", "No"))
+		if(!give_return)
 			return
-		var/initial_outfits = tgui_alert(usr, "Select outfit", "Quick Dress", list("Bluespace Tech", "Show All", "Cancel"))
-		if (initial_outfits == "Cancel")
+
+	if(character_option == "Selected Character")
```

Comment:
Just in case i'd expose the option lists to a var and compare this via list index, not required, just would ben ice to have